Streetcode/Streetcode.XUnitTest/MediatRTests/AdditionalContent/AdditionalContentTests/CoordinateTests/GetCoordinatesByStreetcodeIdHandlerTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore.Query;
+using Moq;
+using Streetcode.BLL.DTO.AdditionalContent.Coordinates;
+using Streetcode.BLL.DTO.AdditionalContent.Coordinates.Types;
+using Streetcode.BLL.MediatR.AdditionalContent.Coordinate.GetByStreetcodeId;
+using Streetcode.DAL.Entities.AdditionalContent.Coordinates.Types;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Streetcode.XUnitTest.MediatRTests.AdditionalContent.CoordinateTests
+{
+    public class GetCoordinatesByStreetcodeIdHandlerTests
+    {
+        [Theory]
+        [InlineData(13)]
+        public async Task GetByStreetcodeId_ReturnsList(int streetcode_id)
+        {
+            //Arrange
+            var coordinates = new List<StreetcodeCoordinate>()
+            {
+                new StreetcodeCoordinate
+                {
+                    Id = 1,
+                    StreetcodeId = 13
+                },
+                new StreetcodeCoordinate
+                {
+                    Id = 2,
+                    StreetcodeId = 13
+                }
+            };
+
+            var mockRepo = new Mock<IRepositoryWrapper>();
+
+           mockRepo.Setup(repo => repo.StreetcodeCoordinateRepository.GetAllAsync(
+               It.IsAny<Expression<Func<StreetcodeCoordinate, bool>>>(),
+               It.IsAny<Func<IQueryable<StreetcodeCoordinate>,
+               IIncludableQueryable<StreetcodeCoordinate, object>>>()))
+               .ReturnsAsync(coordinates);
+            var mockMapper = new Mock<IMapper>();
+
+            mockMapper.Setup(x => x.Map<IEnumerable<StreetcodeCoordinateDTO>>(It.IsAny<IEnumerable<object>>()))
+                .Returns((List<StreetcodeCoordinate> coordinates) =>
+                {
+                    var dtolist = new List<StreetcodeCoordinateDTO>();
+
+                    for (int i = 0; i < coordinates.Count; i++)
+                        dtolist.Add(new StreetcodeCoordinateDTO
+                        {
+                            Id = coordinates[i].Id,
+                            StreetcodeId = coordinates[i].StreetcodeId
+                        });
+
+                    return dtolist;
+                });
```

Comment:
Lambda doesnt work as expected, move logic from return method, use private methods, do we need FOR here?